controllers/replica_set_controller.go

```diff
@@ -461,6 +464,31 @@ func (r *ReplicaSetReconciler) ensureService(mdb mdbv1.MongoDBCommunity) error {
 	return err
 }
 
+func (r *ReplicaSetReconciler) createProcessPortManager(mdb mdbv1.MongoDBCommunity) (*agent.ReplicaSetPortManager, error) {
+	currentAC, err := automationconfig.ReadFromSecret(r.client, types.NamespacedName{Name: mdb.AutomationConfigSecretName(), Namespace: mdb.Namespace})
+	if err != nil {
+		return nil, errors.Errorf("could not read existing automation config: %s", err)
+	}
+
+	sts, err := r.client.GetStatefulSet(mdb.NamespacedName())
+	if err != nil {
+		if !apiErrors.IsNotFound(err) {
+			return nil, fmt.Errorf("failed to get StatefulSet: %s", err)
+		}
+
+		// sts's name is needed for GetAllDesiredMembersPodState logic
+		sts.Namespace = mdb.Namespace
```

Comment:
Should this logic be executed if err != nil?

Reply:
This is executed only if err IsNotFound. But if you're asking I'll look if it can be improved.

Reply:
I have removed it completely.